### PR TITLE
Fold tile and content metadata extensions

### PR DESCRIPTION
**Pull request by j9liu**
This PR folds the `3DTILES_metadata` extension for explicit tile, content, and group metadata. It includes the following changes:

- Added `findTileMetadata.js` (should this be organized into a module with `findContentMetadata` and `findGroupMetadata` ?)
- Updated `findContentMetadata` and `findGroupMetadata` to handle `metadata` fields in the tileset JSON
- Swept through documentation and added entries for 3D Tiles 1.1 changes.
- Updated unit tests
  - Add 1.1 versions of explicit metadata tilesets
  - Reorganize tests such that legacy tests appear under a `3DTILES_metadata` description
  - Something I missed in #10150, I forgot to add tests that assign content metadata to the various `___3DTileContent` classes. Maybe it wasn't necessary, but since they exist for `"assigns group metadata"` it only seemed fair to add for `content.metadata` too

cc @ptrgags for review

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @j9liu!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Review by ptrgags — COMMENTED** [review]
@j9liu I only did a light first pass review so far. I noticed a way that we can avoid duplicating so many binary files

File: Source/Scene/findGroupMetadata.js [hunk 1]
```diff
@@ -13,11 +15,13 @@ import hasExtension from "./hasExtension.js";
...
+  if (!defined(tileset.metadata)) {
+    return undefined;
   }
 
-  return undefined;
+  const group = hasExtension(contentHeader, "3DTILES_metadata")
+    ? contentHeader.extensions["3DTILES_metadata"].group
+    : contentHeader.group;
+
+  return tileset.metadata.groups[group];
```
[the remark this answers is not in the document]
Comment (j9liu) [reply Source/Scene/findGroupMetadata.js:26]: I don't think this is the same thing -- `tileset` is the actual `Cesium3DTileset` with the schema loaded in.

**Comment by j9liu** [comment]
@ptrgags updated with a squashed commit!

File: Specs/Scene/Batched3DModel3DTileContentSpec.js [hunk 2]
```diff
@@ -487,6 +488,30 @@ describe(
         class: metadataClass,
       });
 
+      const contentMetadataClass = new MetadataClass({
```
Comment (ptrgags) [inline Specs/Scene/Batched3DModel3DTileContentSpec.js:491]: Could we move these into a `beforeAll()` block like we've done before in past PRs?

File: Specs/Scene/Geometry3DTileContentSpec.js [hunk 3]
```diff
@@ -826,6 +827,30 @@ describe(
         class: metadataClass,
       });
 
+      const contentMetadataClass = new MetadataClass({
```
Comment (ptrgags) [inline Specs/Scene/Geometry3DTileContentSpec.js:830]: Same thing here and in other specs you updated, move to a `beforeAll()` block

File: Specs/Scene/findTileMetadataSpec.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import {
+  Cartesian3,
+  findTileMetadata,
+  MetadataClass,
+} from "../../Source/Cesium.js";
+
+describe("Scene/findTileMetadata", function () {
+  const tileClass = new MetadataClass({
```
Comment (ptrgags) [inline Specs/Scene/findTileMetadataSpec.js:8]: Same thing here, please put the constructor call in `beforeAll()`

**Review by ptrgags — COMMENTED** [review]
@j9liu looks good, just noticed that some of the unit test setup could be moved to `beforeAll()` blocks in several files (make sure to sweep through the diff for others)

**Comment by j9liu** [comment]
@ptrgags updated!

**Comment by ptrgags** [comment]
Looks good now, thanks @j9liu!

**Pull request merged**